src/reducers/SideBarReducer.js

```diff
@@ -1,16 +1,58 @@
-import { Map } from 'immutable';
+import { Map, List } from 'immutable';
 
 const INITIAL_STATE = Map({
-    currentSelectedCity: 'All cities',
+    'sideBarShown': true,
+    'currentSelectedCity': 'All cities',
+    'currentSelectedCategories': List(['All categories']),
```

Comment:
It feels a littttle weird to me that `'All categories'` is a category, instead of like `['A', 'B', 'C']` representing "all categories". What are the tradeoffs?

Reply:
1. When the backend filter posts, if the category selected is all categories, then I immediately give all the posts I have instead of having to loop through each post for the right category posts. (but for this point, i guess I can just simply check if the selected_category_list length and see if all is selected). 
2. if I simply allow the backend to recognize 'All categories', then I don't need to have another algorithm to put all the category names in the selected_category_list

I think the tradeoff is very little. Both works just fine.